pkg/openshift/rosa/rosa.go

```diff
@@ -175,29 +175,41 @@ func getVersion(ctx context.Context, rosaBinary string) (string, error) {
 }
 
 // verifyLogin validates the authentication details provided are valid by logging in with rosa cli
-func verifyLogin(ctx context.Context, rosaBinary string, token string, ocmEnvironment ocmclient.Environment, awsCredentials *awscloud.AWSCredentials) error {
-	commandArgs := []string{
-		"login",
-		"--token", token,
-		"--env", string(ocmEnvironment),
-	}
+func verifyLogin(ctx context.Context, rosaBinary string, token string, clientID string, clientSecret string, ocmEnvironment ocmclient.Environment, awsCredentials *awscloud.AWSCredentials) error {
+	commandArgs := []string{"login"}
 
 	command := exec.CommandContext(ctx, rosaBinary, commandArgs...)
 	command.Env = append(command.Environ(), awsCredentials.CredentialsAsList()...)
-	command.Env = append(command.Env, fmt.Sprintf("OCM_CONFIG=%s/ocm.json", os.TempDir()))
 
-	_, _, err := cmd.Run(command)
+	if clientID != "" && clientSecret != "" {
+		command.Args = append(command.Args, "--client-id", clientID)
+		command.Args = append(command.Args, "--client-secret", clientSecret)
+		command.Args = append(command.Args, "--govcloud")
+		// TODO: Work around. The rosa cli for govcloud does not support the --env passing the api endpoint.
+		// The environment selection can be handled with a data structure that maps the environment to the api endpoint.
+		if ocmEnvironment == "https://api.int.openshiftusgov.com" {
+			ocmEnvironment = "integration"
+		}
+	} else {
+		command.Args = append(command.Args, "--token", token)
+		command.Env = append(command.Env, fmt.Sprintf("OCM_CONFIG=%s/ocm.json", os.TempDir()))
+	}
+	command.Args = append(command.Args, "--env", string(ocmEnvironment))
+	command.Args = append(command.Args, "--region", string(awsCredentials.Region))
+
+	_, stderr, err := cmd.Run(command)
 	if err != nil {
-		return fmt.Errorf("login failed %v", err)
+		return fmt.Errorf("login failed %v, rosa output %v", err, stderr)
```

Comment:
yeah this is fine, let's just rewrite the error slightly so it makes more sense when reading it

```suggestion
		return fmt.Errorf("login failed with %q: %w", stderr, err)
```